lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -86,20 +89,24 @@ class UnusedJavaScript extends ByteEfficiencyAudit {
     } = context.options || {};
 
     const items = [];
-    for (const [url, scriptCoverages] of Object.entries(artifacts.JsUsage)) {
-      const networkRecord = networkRecords.find(record => record.url === url);
+    for (const [scriptId, scriptCoverage] of Object.entries(artifacts.JsUsage)) {
+      const script = artifacts.Scripts.find(s => s.scriptId === scriptId);
+      if (!script) continue; // This should never happen.
+
+      const networkRecord = networkRecords.find(record => record.url === script.url);
```

Comment:
remember: `url` for inline scripts is the HTML element, so this still works as expected wrt: estimating the transfer size from the given network record